File: CSharp/1410.cs
using System; 
using System.Linq;


class URI {

    static void Main(string[] args) 
    {   
        int[] lista;
        int[] a;
        int[] d;
        bool  impedido=false;

        while (true) 
        {
            lista = Console.ReadLine().Split().Select(int.Parse).ToArray();
            if(lista[0]==0 && lista[1]==0)
            {break;}

            a=Console.ReadLine().Split().Select(int.Parse).OrderByDescending(x=>x).ToArray();
            d=Console.ReadLine().Split().Select(int.Parse).OrderByDescending(x=>x).ToArray();

            foreach(int i in a)
            {
                if((i<d[d.Length-2] && i<d[d.Length-1]) ||(i<d[d.Length-2]))
                {
                    Console.WriteLine("Y");
                    impedido =true;
                    break;
                }
                
            }
            if(!impedido)
            {Console.WriteLine("N");}
            impedido=false;
        }
    }
}

File: Python/1410.py
impedido = False
while True:
    lista = list(map(int, input().split()))
    if lista[0] == 0 and lista[1] == 0:
        break

    a = sorted(map(int, input().split()), reverse=True)
    d = sorted(map(int, input().split()), reverse=True)
    for i in a:
        if (i < d[-2] and i < d[-1]) or (i < d[-2]):
            print("Y")
            impedido = True
            break


    if not impedido:
        print("N")
        
    impedido = False
